src/components/Form/index.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react'
+import dot from 'dot-object'
+
+export default class Form extends Component {
+   static propTypes = {
+      children: React.PropTypes.node,
+      onError: React.PropTypes.func,
+      onSubmit: React.PropTypes.func
+   }
+
+   constructor(props) {
+      super(props)
+
+      this.handleSubmit = this.handleSubmit.bind(this)
+
+      this.initialValuesLoaded = false
+
+      // Setup state
+      this.state = {}
+      props.children.forEach((child) => {
+         if (this.childIsRelevant(child)) {
+            this.state[child.props.name] = {
+               value: '',
+               meta: {}
+            }
+         }
+      })
+   }
+
+   componentWillReceiveProps(props) {
+      let initialValues = props.initialValues
+
+      if (initialValues && !this.initialValuesLoaded) {
+         const newState = {}
+
+         initialValues = dot.dot(initialValues)
+
+         props.children.forEach((child) => {
+            if (this.childIsRelevant(child)) {
+               const value = initialValues
+                  ? initialValues[child.props.name] || ''
+                  : ''
+               newState[child.props.name] = {
+                  value,
+                  meta: {
+                     touched: false,
+                     error: this.checkError(child, value, props)
+                  }
+               }
+            }
+         })
+
+         this.initialValuesLoaded = true
+         this.setState(newState)
+      }
+   }
+
+   getTestFromString(str) {
+      switch (str) {
+         case 'required':
+            return value => !!value
+         case 'email':
+            return (value) => {
+               const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ // eslint-disable-line
+               return regex.test(value)
+            }
+         default:
+            break
+      }
+   }
+
+   checkError(child, initialValue, initialProps) {
+      const value = initialValue && typeof initialValue === 'string'
+         ? initialValue
+         : (this.state[child.props.name] && this.state[child.props.name].value) || ''
+      const props = initialProps || this.props
+
+      const validations = props.validations && props.validations[child.props.name]
+      if (!validations) return ''
+
+      for (let i = 0; i < validations.length; i += 1) {
+         const validation = validations[i]
+         const test = typeof validation.test === 'string'
+            ? this.getTestFromString(validation.test)
+            : validation.test
+         if (!test(value)) {
```

Comment:
If line 84 executes, then isn't the value of `test` a boolean (or `undefined`)? If so, how does this execute? Sorry I might be missing something super obvious here